### PR TITLE
Adds ETF(...) to UniverseDefinitions

### PR DESCRIPTION
**Pull request by gsalaz98**
#### Description

Adds shortcut to allow adding universes with pre-defined universe functions, e.g. `Universe.ETF(Symbol, ...)`

Various `UniverseDefinitions` members were axed in this PR, and as such might cause breaking changes for users using QC500 universe explicitly.

Adds ETF constituents universe framework regression algorithm (C#/Python)
#### Related Issue

N/A
#### Motivation and Context

Improve developer ergonomics
#### Requires Documentation Change

Yes, QC500 is gone and Universe.ETF has been added
#### How Has This Been Tested?

Modified regression algorithms to use new shortcut with AddUniverse and added new regression algorithms using Framework
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [ ] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm/UniverseDefinitions.cs [hunk 1]
```diff
@@ -23,35 +27,160 @@ namespace QuantConnect.Algorithm
...
+        /// <param name="universeFilterFunc">Function to filter universe results</param>
+        /// <returns>New index constituents Universe</returns>
+        public Universe Index(
+            string indexTicker,
+            string market = null,
+            UniverseSettings universeSettings = null,
+            Func<IEnumerable<ETFConstituentData>, IEnumerable<Symbol>> universeFilterFunc = null)
+        {
+            return Index(
+                Symbol.Create(indexTicker, SecurityType.Index, market),
```
Comment (Martin-Molinero) [inline Algorithm/UniverseDefinitions.cs:134]: The overloads accepting a string ticker throw a null reference exception, market can't be null. Let's add a tests covering the cases too

**Review by Martin-Molinero — CHANGES_REQUESTED** [review]
Cool! A few test cases still missing 👍

**Review by Martin-Molinero — APPROVED** [review]
Thanks!

File: Algorithm.Python/QuantConnect.Algorithm.Python.csproj [hunk 2]
```diff
@@ -140,6 +139,12 @@
     <None Include="TensorFlowNeuralNetworkAlgorithm.py" />
     <None Include="TiingoPriceAlgorithm.py" />
     <None Include="PearsonCorrelationPairsTradingAlphaModelFrameworkAlgorithm.py" />
+    <None Update="UniverseSelectionDefinitionsAlgorithm.py">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <None Update="ConstituentsQC500GeneratorAlgorithm.py">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (Martin-Molinero) [inline Algorithm.Python/QuantConnect.Algorithm.Python.csproj:146]: nit - maybe we can go back to existing implementation regarding the imports? seems we didn't use `CopyToOutputDirectory`

File: Algorithm.CSharp/ETFConstituentUniverseRSIAlphaModelAlgorithm.cs [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    /// Example algorithm demonstrating the usage of the RSI indicator
+    /// in combination with ETF constituents data to replicate the weighting
+    /// of the ETF's assets in our own account.
+    /// </summary>
+    public class ETFConstituentUniverseRSIAlphaModelAlgorithm : QCAlgorithm
+    {
+        /// <summary>
+        /// Store constituents data when doing universe selection for later use in the alpha model
+        /// </summary>
+        public List<ETFConstituentData> Constituents = new List<ETFConstituentData>();
```
Comment (Martin-Molinero) [inline Algorithm.CSharp/ETFConstituentUniverseRSIAlphaModelAlgorithm.cs:38]: Believe this information will be available through the cache of the underlying see https://github.com/QuantConnect/Lean/blob/master/Common/Securities/SecurityCache.cs#L410 and it's usages, would be a nice example usage of it 👍

File: Algorithm.CSharp/ETFConstituentUniverseRSIAlphaModelAlgorithm.cs [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+                .Select(x => x.Symbol)
+                .ToList();
+        }
+
+        /// <summary>
+        /// no-op
+        /// </summary>
+        public override void OnData(Slice data)
+        {
+        }
```
Comment (Martin-Molinero) [inline Algorithm.CSharp/ETFConstituentUniverseRSIAlphaModelAlgorithm.cs:81]: Don't think this is required?
`.ToList();` in `return Constituents.Select(x => x.Symbol).ToList();` not required either?
The cleaner & simpler the better 👍

File: Algorithm.CSharp/ETFConstituentUniverseRSIAlphaModelAlgorithm.cs [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+                        // First time we're initializing the RSI.
+                        // It won't be ready now, but it will be
+                        // after 7 data points.
+                        var constituent = constituents[bar.Symbol];
+                        rsiData = new SymbolData(bar.Symbol, constituent, 7);
+                        _rsiSymbolData[bar.Symbol] = rsiData;
+                    }
+
+                    // Let's make sure all RSI indicators are ready before we emit any insights.
+                    allReady = allReady && rsiData.Rsi.Update(new IndicatorDataPoint(bar.Symbol, bar.EndTime, bar.Close));
```
Comment (Martin-Molinero) [inline Algorithm.CSharp/ETFConstituentUniverseRSIAlphaModelAlgorithm.cs:130]: I think could avoid the need to update the RSI manually & waiting for a warmup if used the `algo.RSI()` ?

**Review by Martin-Molinero — APPROVED** [review]

**Comment by jaredbroad** [comment]
🎉🎉

**Pull request merged**